### PR TITLE
Feat: Add Error Event Queue

### PR DESCRIPTION
**Pull request by ArnavG-it**
Features:
- 
- New ErrorQueue ADT for storing CANWrapper_ErrorInfo items, based on CANQueue implementation

Changes:
- 
- Callbacks for the TX timeout errors now run after the TX cache is cleared by looping through the ErrorQueue
- `CANWrapper_Poll_Events()` is now split into `CANWrapper_Poll_Messages()` and `CANWrapper_Poll_Errors()` which can be called separately 
- Timeout detection conditional is broken into steps for readability

Fixes:
- 
- Correctly determine the timeout tick by multiplying `PERIOD_TICKS` by 16 repeats

### CODE REVIEW

**Review by Koloss0 — CHANGES_REQUESTED** [review]

File: Inc/tuk/can_wrapper/error_queue.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @author Om Sevak <om.sevak@umsats.ca>
+ * @author Arnav Gupta <arnav.gupta@umsats.ca>
+ *
+ * @date August 3, 2024
+ */
+
+#ifndef CAN_WRAPPER_MODULE_INC_ERROR_QUEUE_H_
+#define CAN_WRAPPER_MODULE_INC_ERROR_QUEUE_H_
+
+#include "can_wrapper.h"
```
[the remark this answers is not in the document]
Comment (ArnavG-it) [reply Inc/tuk/can_wrapper/error_queue.h:16]: Should `can_wrapper.h` include `error_queue.h` then to use the structure?

[the remark this answers is not in the document]
Comment (Koloss0) [reply Inc/tuk/can_wrapper/error_queue.h:16]: Yep. If that bothers you, you can also split it into `error_info.h` and `error_queue.h`.

[the remark this answers is not in the document]
Comment (ArnavG-it) [reply Inc/tuk/can_wrapper/error_queue.h]: I'll leave it in `error_queue.h` since it makes sense for now.

**Review by Koloss0 — APPROVED** [review]
all set!

**Pull request merged**